### ISSUE TITLE
TEST-50-DISSECT fails

### ISSUE THREAD

**Issue by yuwata** [systemd/systemd#23174]
```
Apr 23 02:25:06 testsuite-50.sh[591]: + VERITY_UUID=f24c3487-6220-1202-adf8-0273a4078f1d
Apr 23 02:25:06 testsuite-50.sh[747]: + systemd-dissect --json=short --root-hash d654c1d52dba3a115ca618f5b11b375ff24c348762201202adf80273a4078f1d /tmp/tmp.ksbDeE/minimal_0.gpt
Apr 23 02:25:06 testsuite-50.sh[748]: + grep -q '{"rw":"ro","designator":"root","partition_uuid":"d654c1d5-2dba-3a11-5ca6-18f5b11b375f","partition_label":"Root Partition","fstype":"squashfs","architecture":"x86-64","verity":"yes",'
Apr 23 02:25:06 testsuite-50.sh[747]: Opened '/tmp/tmp.ksbDeE/minimal_0.gpt' in O_RDONLY access mode, with O_DIRECT disabled (O_DIRECT was requested but not supported).
Apr 23 02:25:06 testsuite-50.sh[747]: Failed to enable O_DIRECT for backing file descriptor for loopback device. Continuing without.
Apr 23 02:25:06 testsuite-50.sh[747]: loop0: Skipping non-partition child (devtype=disk).
Apr 23 02:25:06 kernel: loop0: detected capacity change from 0 to 7188480
Apr 23 02:25:06 kernel:  loop0: p1 p2 p3
Apr 23 02:25:06 testsuite-50.sh[747]: Successfully acquired /dev/loop0, devno=7:0, nr=0, diskseq=0
Apr 23 02:25:06 testsuite-50.sh[747]: Kernel was quicker than us in adding partition 1.
Apr 23 02:25:06 testsuite-50.sh[747]: Kernel was quicker than us in adding partition 2.
Apr 23 02:25:06 testsuite-50.sh[747]: Kernel was quicker than us in adding partition 3.
Apr 23 02:25:06 testsuite-50.sh[747]: Probed fstype 'squashfs' on partition /dev/loop0p1.
Apr 23 02:25:06 testsuite-50.sh[747]: Successfully forked off '(sd-dissect)' as PID 750.
Apr 23 02:25:06 testsuite-50.sh[750]: Mounting /dev/loop0p1 (squashfs) on /tmp/dissect-T4krGI (MS_RDONLY|MS_NODEV "")...
Apr 23 02:25:06 testsuite-50.sh[750]: Failed to mount /dev/loop0p1 (type squashfs) on /tmp/dissect-T4krGI (MS_RDONLY|MS_NODEV ""): Block device required
Apr 23 02:25:06 testsuite-50.sh[750]: Failed to mount dissected image: Block device required
Apr 23 02:25:06 kernel: /dev/loop0p1: Can't open blockdev
Apr 23 02:25:06 testsuite-50.sh[747]: Failed to read /etc/hostname of image: No such file or directory
Apr 23 02:25:06 testsuite-50.sh[747]: /etc/machine-id file of image is empty.
Apr 23 02:25:06 testsuite-50.sh[747]: Failed to read has-init-system boolean: Input/output error
Apr 23 02:25:06 testsuite-50.sh[747]: (sd-dissect) failed with exit status 1.
Apr 23 02:25:06 testsuite-50.sh[747]: Failed to acquire image metadata: Block device required
```
[Full journal is here.](https://jenkins-systemd.apps.ocp.ci.centos.org/job/upstream-centos8/7740/artifact/systemd-centos-ci/artifacts_e6zeqszi/testsuite-logs-upstream.gVi/TEST-50-DISSECT/system.journal)

**Comment by bluca** [comment]
Since https://github.com/systemd/systemd/pull/22992 the old failures cannot be seen, but there's a new one appearing every now and then:

```
Apr 30 03:14:00 H systemd[320]: device-mapper: create ioctl on d5815f6c368afc1bef9585f92fca23cfd7ad7a16bcbe7dc5bb9c939a8bfdfb40-verity CRYPT-VERITY-f865959ab953409084f6e74ac1f40456-d5815f6c368afc1bef9585f92fca23cfd7ad7a16bcbe7dc5bb9c939a8bfdfb40-verity failed: Device or resource busy
Apr 30 03:14:00 H systemd[320]: Device d5815f6c368afc1bef9585f92fca23cfd7ad7a16bcbe7dc5bb9c939a8bfdfb40-verity already exists.
Apr 30 03:14:00 H systemd[320]: Failed to wait for device to be initialized: Connection timed out
Apr 30 03:14:02 H kernel: /dev/mapper/d5815f6c368afc1bef9585f92fca23cfd7ad7a16bcbe7dc5bb9c939a8bfdfb40-verity: Can't open blockdev
Apr 30 03:14:02 H systemd[319]: Device d5815f6c368afc1bef9585f92fca23cfd7ad7a16bcbe7dc5bb9c939a8bfdfb40-verity is not active.
```

@poettering any idea?

**Comment by yuwata** [comment]
@mrc0mmand and @bluca IIRC, I have not seen this issue recently. Is this already fixed? Maybe by #23291 or #23352?

**Comment by bluca** [comment]
Interesting - at a glance those don't feel related? But can't say for certain
